lib/Controller/RoomController.php

```diff
@@ -1140,7 +1141,11 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
 		}
 
 		// add the remaining users in batch
-		$this->participantService->addUsers($this->room, $participantsToAdd, $addedBy);
+		try {
+			$this->participantService->addUsers($this->room, $participantsToAdd, $addedBy);
+		} catch (CannotReachRemoteException $e) {
```

Comment:
Could we end up in a situation where only some users were added to the room? Like if the array contains 2 users and the first one throws a `CannotReachRemoteException`, what happens with the second one?

Reply:
the array case is only used for groups and circles atm. For normal users, email addresses and federated users the array will always only be a single entry as you have to send 1 API call per target to add 🙈